tests/e2e_tests/run

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+set -e
+
+COMPOSE_CMD="docker-compose -p gui-tests \
+             -f ${INTEGRATION_PATH}/docker-compose.yml \
+             -f ${INTEGRATION_PATH}/docker-compose.demo.yml \
+             -f tests/e2e_tests/docker-compose.e2e-tests.yml"
+
+# by default just add minio, with COMPOSE_CMD_BASE this creates the standard onprem ST setup
+COMPOSE_FILES_DEFAULT=( "${INTEGRATION_PATH}/docker-compose.storage.minio.yml" )
+COMPOSE_FILES=()
+
+usage() {
+    echo "runner script for gui-specific integration tests"
+    echo ""
+    echo "./gui-tests"
+    echo -e "\t-h --help"
+    echo -e "\t-c --skip-cleanup \tleave containers running after tests"
+    echo -e "\t-f=<FILE>         \tspecify custom compose file(s); default files will not be used,"
+    echo -e "\t                  \tmake sure to specify all files you need"
+    echo -e "\t other args will be passed to the testing container's py.test command"
+    echo ""
+    echo -e "examples:"
+    echo -e "run all tests, default ST setup:"
+    echo -e "\t./run"
+}
+
+parse_args(){
+    whitespace="[[:space:]]"
+
+    # FIXME args must be passed in the form -f=<path>; spaces don't workalthough they're standard...
+    while [ $# -gt 0 ]; do
+        PARAM=`echo $1 | awk -F= '{print $1}'`
+        VALUE=`echo $1 | awk -F= '{print $2}'`
+        case $PARAM in
+            -h | --help)
+            usage
+            exit
+            ;;
+            -c | --skip-cleanup)
+            SKIP_CLEANUP=1
+            ;;
+            -f)
+            COMPOSE_FILES+=( $VALUE )
+            ;;
+            *)
+        esac
+        shift 1
+    done
+
+    make_compose_cmd
+}
+
+make_compose_cmd () {
+    if [ ${#COMPOSE_FILES[@]} -eq 0 ]
+    then
+        COMPOSE_FILES=$COMPOSE_FILES_DEFAULT
+    fi
+
+    for var in "${COMPOSE_FILES[@]}"
+    do
+        COMPOSE_CMD+=" -f ${var}"
+    done
+}
+
+run_tests() {
+    $COMPOSE_CMD build && $COMPOSE_CMD up -d
```

Comment:
One disadvantage of this method is that you can't spin up new containers for a new test, so you need to ensure that:

* All tests need the same set of containers
* No test changes the containers in any way that affects a subsequent test

I can definitely see the advantages to this method also though (reproducibility, ease of setup, etc), so I'm not saying this is bad. But have you given it any thought?

Reply:
I am not sure that I fully understand your question do you mean the mandatory building part or using the composition as a whole?
Regarding the mandatory building:
To make good use of the pipeline I would pass on the build results and mount them in the gui container. same could go for the test container though...
Regarding the composition:
The existing tests basically build on top of each other (one uploading an artifact, the next using this to create a deployment, ...) so they do change the containers + the containers used are the whole demo setup. There also is no mocking/ initialization/ tear down per test, but rather a walkthrough for a new user.

The UI unit tests will happen apart from that in the future and will not be bound to the infrastructure.